src/test/java/service/impl/DataConverterImplTest.java

```diff
@@ -0,0 +1,79 @@
+package service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import model.FruitTransaction;
+import model.Operation;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import service.DataConverter;
+
+class DataConverterImplTest {
+    private static DataConverter dataConverter;
+
+    @BeforeAll
+    static void beforeAll() {
+        dataConverter = new DataConverterImpl();
+    }
+
+    @Test
+    void convertToTransaction_nullList_notOk() {
+        assertThrows(RuntimeException.class, () -> dataConverter.convertToTransaction(null));
+    }
+
+    @Test
+    void convertToTransaction_incorrectDataLine_notOk() {
+        assertThrows(RuntimeException.class, () -> dataConverter.convertToTransaction(
+                List.of("type,fruit,quantity", "b,banana20")));
+        assertThrows(RuntimeException.class, () -> dataConverter.convertToTransaction(
+                List.of("type,fruit,quantity", "b,banana,20 b,apple,100")));
+        assertThrows(RuntimeException.class, () -> dataConverter.convertToTransaction(
+                List.of("type,fruit,quantity", "")));
+        assertThrows(RuntimeException.class, () -> dataConverter.convertToTransaction(
+                List.of("type,fruit,quantity", "b;banana;20")));
+    }
```

Comment:
In one method should use one test case

Reply:
Why? I guess in this case will be better put all this incorrect strings into one method.

Reply:
in general, it's better to divide your test logic into several methods with proper names. But in this case I think it's enough to leave everything as is